src/aws-cpp-sdk-core/include/aws/core/auth/STSCredentialsProvider.h

```diff
@@ -25,7 +25,7 @@ namespace Aws
         class AWS_CORE_API STSAssumeRoleWebIdentityCredentialsProvider : public AWSCredentialsProvider
         {
         public:
-            STSAssumeRoleWebIdentityCredentialsProvider();
+            STSAssumeRoleWebIdentityCredentialsProvider(Aws::Client::ClientConfiguration config = {});
```

Comment:
I'm really sorry for a a big delay in the handling of this PR.
I've been running a CI check and just noticed that

it is actually changing the default constructor to make an IMDS call (to resolve a region, not always though, but still...), this is a significant change of existing behavior.

could you please change it to
```
STSAssumeRoleWebIdentityCredentialsProvider(Aws::Client::ClientConfiguration config = {Aws::Client::ClientConfigurationInitValues{true}});
```
so IMDS won't be auto-enabled?

We should be able to merge the PR shortly after this.
Thank you!

Reply:
Done.